iothub_client/samples/iothub_client_sample_upload_to_blob_custom/iothub_client_sample_upload_to_blob_custom.c

```diff
@@ -0,0 +1,197 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+// CAVEAT: This sample is to demonstrate azure IoT client concepts only and is not a guide design principles or style
```

Comment:
I would also add why this sample is a lot better than the original:
1. How to get SAS token (usable by external tools such as az_copy).
2. How to use the in-box Blob new upload client.
3. How to retry blob upload _per block_ (as opposed to the entire file).

Reply:
Added.